kedro-datasets/tests/conftest.py

```diff
@@ -5,9 +5,79 @@
 https://docs.pytest.org/en/latest/fixture.html
 """
 
+from typing import Callable
+from unittest.mock import MagicMock
+
+import aiobotocore.awsrequest
+import aiobotocore.endpoint
+import aiohttp
+import aiohttp.client_reqrep
+import aiohttp.typedefs
+import botocore.awsrequest
+import botocore.model
 from kedro.io.core import generate_timestamp
 from pytest import fixture
 
+BUCKET_NAME = "test_bucket"
+IP_ADDRESS = "127.0.0.1"
+PORT = 5555
+ENDPOINT_URI = f"http://{IP_ADDRESS}:{PORT}/"
+
+
+"""
+Patch aiobotocore to work with moto
+See https://github.com/aio-libs/aiobotocore/issues/755
+"""
+
+
+class MockAWSResponse(aiobotocore.awsrequest.AioAWSResponse):
+    def __init__(self, response: botocore.awsrequest.AWSResponse):
+        self._moto_response = response
+        self.status_code = response.status_code
+        self.raw = MockHttpClientResponse(response)
+
+    # adapt async methods to use moto's response
+    async def _content_prop(self) -> bytes:
+        return self._moto_response.content
+
+    async def _text_prop(self) -> str:
+        return self._moto_response.text
+
+
+class MockHttpClientResponse(aiohttp.client_reqrep.ClientResponse):
+    def __init__(self, response: botocore.awsrequest.AWSResponse):
+        async def read(self, n: int = -1) -> bytes:
+            # streaming/range requests. used by s3fs
+            return response.content
```

Comment:
Should this be in here? I would have thought it would be in `MockHttpClientResponse` and not in the `__init__`.

Reply:
Good question, tbh I didn't try other variations of this specific patch. I used the most recent and clean snippet I could find in the threads here https://github.com/aio-libs/aiobotocore/issues/755 https://gist.github.com/giles-betteromics/12e68b88e261402fbe31c2e918ea4168, assuming that writing my own wasn't going to be better than all the iterations from the community.